File: src/domain/entities/OrderItem.js
const Money = require("../value-objects/Money");
const Quantity = require("../value-objects/Quantity");

class OrderItem {
  constructor(menuId, menuName, price, quantity) {
    this.menuId = menuId;
    this.menuName = menuName;
    this.price = new Money(price);
    this.quantity = new Quantity(quantity);
  }

  total() {
    return this.price.value * this.quantity.value;
  }
}

module.exports = OrderItem;

File: src/domain/entities/OrderItem.py
from src.domain.value_objects.Money import Money
from src.domain.value_objects.Quantity import Quantity

class OrderItem:
    def __init__(self, menu_id: str, menu_name: str, price: int, quantity: int):
        self.menu_id = menu_id
        self.menu_name = menu_name
        self.price = Money(price)
        self.quantity = Quantity(quantity)

    def total(self):
        return self.price.value * self.quantity.value
